core/src/test/scala/org/typelevel/paiges/PaigesScalacheckTest.scala

```diff
@@ -189,26 +189,21 @@ class PaigesScalacheckTest extends FunSuite {
   }
   test("a.flatten == a.flatten.flatten") {
     forAll { (a: Doc) =>
-      val aflat = a.flatten
-      assert(aflat === aflat.flatten)
-    }
-  }
-  test("a.flatten == a.flattenOption.getOrElse(a)") {
```

Comment:
this law would be nice to add back with the tri-state FlattenResult I suggest above.

basically: `a.flatten.toDoc === a.flatten.toDoc.flatten.toDoc` which is what you have, but also:
`a.flatten.toDoc === a.flatten.orElse(a)` and also `val fa = a.flatten; if (fa.wasFlattened) fa.toDoc =!= a else fa.toDoc === a`

So, I think we want to exercise all three results of a flattening.